File: code/系統程式/irvm/rv0/fact.c
// #include <stdio.h>

// 遞回計算階乘
long long factorial(int n) {
    // 終止條件：0! = 1, 1! = 1
    if (n <= 1) {
        return 1;
    }
    // 遞回呼叫：n! = n * (n-1)!
    return n * factorial(n - 1);
}

int main() {
    int number = 5;
    long long result = factorial(number);
    
    // 注意：在裸機 (Bare-metal) 環境下可能無法使用 printf
    // 若是在 Linux 環境則正常運作
    // printf("Factorial of %d is %lld\n", number, result);
    
    return result;
}

File: code/系統程式/interpreter/prolog/pl/fact.pl
% 終止條件：0! = 1
factorial(0, 1).

% 遞迴規則
factorial(N, F) :-
    N > 0,
    N1 is N - 1,
    factorial(N1, F1),
    F is N * F1.

% 查詢範例：
% ?- factorial(5, W).  % W = 120